### PR TITLE
Run 1 reco tuning

### PR DESCRIPTION
**Pull request by rovere**
This PR is meant to reproduce Run1-like tracking in CMSSW 75X. There are caveats, of course, and you should not expect to have **exactly** the same reconstruction code that we had in 53.
- All the architectural changes have been kept, like the sane usage of EDProducers instead of the insane use of ES quantities.
- In general all C++ related changes have been kept
- JetCore iteration and ClusterSplitting have been removed
- The order of the iterations has been restored
- The Muon specific iterations have been kept but, if you want, you can use the `earlyGeneralTracks` that are produced **before** running these iterations
- The usage of MVA has been disabled and the old selector cuts have been restored, taken from 5_3_27 (module human errors in diff-ing configuration files)
- PixelVertices and PixelTracks have been both restored

In order to use the Run1-like tracking, please either add:

```
--customise RecoTracker/Configuration/customiseForRunI.customiseForRunI
```

to `cmsDriver.py` command, or:

```
from RecoTracker.Configuration.customiseForRunI import customiseForRunI

#call to customisation function customiseForRunI imported from RecoTracker.Configuration.customiseForRunI
process = customiseForRunI(process)
```

directly into your final python configuration file.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
We need to discuss if this stays as an experimental customize or if it becomes a regularly tested setup (with corresponding relvals etc).
Also, if it's beyond experimental setup, it will need to be made era-aware

**Comment by rovere** [comment]
Ciao Slava,
I agree. Let's first see if this turns out to be somehow useful to anybody. If yes I can make it era-aware, with some guidance.
As for testing, unless you activate it, I hope there will be no changes :smiley:

File: RecoTracker/Configuration/python/customiseForRunI.py [hunk 1]
```diff
@@ -33,11 +46,59 @@ def customiseForRunI(process):
...
+    # Now get rid of any pre-splitting business
+    process.siPixelClusters = process.siPixelClustersPreSplitting.clone()
+    process.pixeltrackerlocalreco.replace(process.siPixelClustersPreSplitting, process.siPixelClusters)
+    process.pixeltrackerlocalreco.replace(process.siPixelRecHitsPreSplitting, process.siPixelRecHits)
+    process.clusterSummaryProducer.pixelClusters = 'siPixelClusters'
+    process.globalreco.replace(process.MeasurementTrackerEventPreSplitting, process.MeasurementTrackerEvent)
+    process.globalreco.replace(process.siPixelClusterShapeCachePreSplitting, process.siPixelClusterShapeCache)
+
+    # Now restore pixelVertices wherever was not possible with an ad-hoc RunI cfg
+    process.muonSeededTracksInOutSelector.vertices = 'pixelVertices'
```
Comment (slava77) [inline RecoTracker/Configuration/python/customiseForRunI.py]: these were not in RunI (53X).

Comment (rovere) [inline RecoTracker/Configuration/python/customiseForRunI.py]: Ciao Slava,
I am not sure what you mean here: which line are you referring to, exactly?
As I mentioned in the comment, this is **not** meant to be a 1:1 replica of RunI tracking, but the best approximation we could have, including all C++ and class layout changes that have been done in between.

Comment (slava77) [inline RecoTracker/Configuration/python/customiseForRunI.py]: I mean the muonSeededTracksInOutSelector

Comment (rovere) [inline RecoTracker/Configuration/python/customiseForRunI.py]: Ciao Slava,
gettig rid of them would have been more work than added value, I fear. As I wrote in the initial comment, if you want to **ignore** them, you can use the earlyGeneralTrack collection, which is produced before running the muon seeded iterations.

Comment (slava77) [inline RecoTracker/Configuration/python/customiseForRunI.py]: That's fine, I understand.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/8691	ed0a3c1
I checked (in CMSSW_7_5_0_pre2 /test area test8691/) that it runs, to see what the impact is for the PU35@25ns scenario.

The total time per event increase from 17 s to 28 s. 
- about + 10 s is from increase in time in tobTecStepTrackCandidates and pixelLessStepTrackCandidates
- a few other modules (pf, muon etc) driven by track multiplicity account for +1 -- +2 s
- jetcore and other removed sequences account for 1.9 s

no surprises here

**Comment by davidlange6** [comment]
+1

**Pull request merged**